src/k2/dto/K23SIInspect.h

```diff
@@ -0,0 +1,97 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
+
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in all copies
+    or
+    substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS",
+    WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+    DAMAGES OR OTHER
+    LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+#pragma once
+
+#include <k2/common/Common.h>
+#include <k2/module/k23si/TxnManager.h>
+#include <k2/transport/PayloadSerialization.h>
+#include <k2/transport/Status.h>
+
+namespace k2 {
+namespace dto {
+
+// For test and debug, requests all versions including WIs for a particular key
+// without affecting transaction state
+struct K23SIInspectRecordsRequest {
+    // These fields make the request compatible with the PartitionRequest wrapper
+    Partition::PVID pvid; // the partition version ID. Should be coming from an up-to-date partition map
+    String collectionName;
+    Key key; // the key to gather all records for
+    K2_PAYLOAD_FIELDS(pvid, collectionName, key);
+};
+
+struct K23SIInspectRecordsResponse {
+    std::vector<DataRecord> records;
```

Comment:
I assume this is the DataRecord from 3si module?

Reply:
Yes. Are you worried about the header dependency?

Reply:
yes - are you moving DataRecord to DTO?

Reply:
ok, I moved it